src/utils/helpers.ts

```diff
@@ -154,49 +142,33 @@ export const getLatestDiscussionCommentId = (
     .reduce((a, b) => (a.node.createdAt > b.node.createdAt ? a : b))?.node
     .databaseId;
 
-export const getCommentId = (url: string) =>
-  /comments\/(?<id>\d+)/g.exec(url)?.groups?.id;
-
-export async function openInBrowser(
+export async function generateGitHubWebUrl(
   notification: Notification,
   accounts: AuthState,
 ) {
-  if (notification.subject.type === 'Release') {
-    getReleaseTagWebUrl(notification, accounts.token).then(({ url }) =>
-      openExternalLink(
-        generateGitHubWebUrl(
-          url,
-          notification.id,
-          accounts.user?.id,
-          undefined,
-        ),
-      ),
-    );
-  } else if (notification.subject.type === 'Discussion') {
-    getDiscussionUrl(notification, accounts.token).then(
-      ({ url, latestCommentId }) =>
-        openExternalLink(
-          generateGitHubWebUrl(
-            url || `${notification.repository.url}/discussions`,
-            notification.id,
-            accounts.user?.id,
-            latestCommentId
-              ? '#discussioncomment-' + latestCommentId
-              : undefined,
-          ),
-        ),
-    );
-  } else if (notification.subject.url) {
-    const latestCommentId = getCommentId(
+  let url = notification.repository.html_url;
+
+  if (notification.subject.latest_comment_url) {
+    url = await getHtmlUrl(
       notification.subject.latest_comment_url,
+      accounts.token,
     );
-    openExternalLink(
-      generateGitHubWebUrl(
-        notification.subject.url,
-        notification.id,
-        accounts.user?.id,
-        latestCommentId ? '#issuecomment-' + latestCommentId : undefined,
-      ),
-    );
+  } else if (notification.subject.url) {
+    url = await getHtmlUrl(notification.subject.url, accounts.token);
+  } else if (notification.subject.type === 'Discussion') {
```

Comment:
What would you think about putting the above conditionals under a conditional for `notification.subject.type`, too? Then it would be a clear hierarchy of: for a discussion, do (logic), for a comment, do (other logic).

Reply:
Moved this to a swtich statement and added a comment explaining the special handling block